scripts/upgrade.ts

```diff
@@ -0,0 +1,254 @@
+
+// v1 migration code
+interface Patch {
+  from: {
+    path: string;
+    content: string;
+  };
+  to: {
+    path: string;
+    content: string;
+  };
+}
+interface Delete {
+  path: string;
+}
+type FileMod = Patch | Delete;
+class UpgradeError extends Error {
+  constructor(err: string) {
+    super(err);
+  }
+}
+const updateImportMap = async (
+  liveVersion: string,
+  stdVersion: string,
+): Promise<Patch> => {
+  const importMapFile = (deno.importMap ?? "./import_map.json").replace(
+    "./",
+    "",
+  );
+
+  const importMapPath = join(Deno.cwd(), importMapFile);
+  if (!(await exists(importMapPath))) {
+    throw new UpgradeError(
+      `${importMapPath} is required to upgrade live dependency verson`,
+    );
+  }
+  const importMapStr = await Deno.readTextFile(importMapPath);
+  const { imports, ...rest }: { imports: Record<string, string> } = JSON.parse(
+    importMapStr,
+  );
+  return {
+    from: {
+      path: importMapPath,
+      content: importMapStr,
+    },
+    to: {
+      path: importMapPath,
+      content: JSON.stringify(
+        {
+          ...rest,
+          imports: {
+            ...imports,
+            "$live/": `https://denopkg.com/deco-cx/live@${liveVersion}/`,
+            "deco-sites/std/":
+              `https://denopkg.com/deco-sites/std@${stdVersion}/`,
+          },
+        },
+        null,
+        2,
+      ),
+    },
+  };
+};
+
+const siteId = async (): Promise<number | undefined> => {
+  const middleware = join(Deno.cwd(), "routes", "_middleware.ts");
+  if (!(await exists(middleware))) {
+    return undefined;
+  }
+  const reg = /siteId: (?<siteId>\d+)/gm;
+  const match = reg.exec(await Deno.readTextFile(middleware));
+  if (!match?.groups) {
+    return undefined;
+  }
+  return +match.groups["siteId"];
+};
+
+const updateDevTsImports = async () => {
+  const devTs = join(Deno.cwd(), "dev.ts");
+  if (!(await exists(devTs))) {
+    throw new UpgradeError(`${devTs} not found`);
+  }
+  const devTsContent = await Deno.readTextFile(devTs);
+  const updateFreshGen = devTsContent.replaceAll(
+    "fresh.gen.ts",
+    "live.gen.ts",
+  );
+  const id = await siteId();
+  const finalContent = id
+    ? updateFreshGen.replace(
+      `await dev(import.meta.url, "./main.ts", {\n`,
+      `await dev(import.meta.url, "./main.ts", {\n  siteId: ${id},\n`,
+    )
+    : updateFreshGen;
+
+  return {
+    from: {
+      path: devTs,
+      content: devTsContent,
+    },
+    to: {
+      path: devTs,
+      content: finalContent,
+    },
+  };
+};
+
+const addMainTsLiveEntrypoint = async () => {
+  const mainTs = join(Deno.cwd(), "main.ts");
+  if (!(await exists(mainTs))) {
+    throw new UpgradeError(`${mainTs} not found`);
+  }
+  const mainTsContent = await Deno.readTextFile(mainTs);
+
+  return {
+    from: {
+      path: mainTs,
+      content: mainTsContent,
+    },
+    to: {
+      path: mainTs,
+      content: mainTsContent.replace(
+        `import manifest from "./fresh.gen.ts";\n`,
```

Comment:
FYI: Might not work on Windows projects, not sure

Reply:
I think it will because it is how deno interprets imports, should not be related to file system separators